Library/Homebrew/upgrade.rb

```diff
@@ -42,6 +43,14 @@ def upgrade_formulae(
         end
       end
 
+      dependency_graph = Utils::TopologicalHash.graph_package_dependencies(formulae_to_install)
+      begin
+        formulae_to_install = dependency_graph.tsort & formulae_to_install
+      rescue TSort::Cyclic
+        # Failed to sort formulae topologically because there are cyclic
+        # dependencies. Let FormulaInstaller handle it.
```

Comment:
Would be good to at least print this for `homebrew_developer?` and/or use `odebug` here. Do we need to handle this in any other way or will it work even if this fails?

Reply:
For developers, `FormulaInstaller` actually raises an error if there are cyclic dependencies, so we technically could raise an error here to save time.

For non-developers, I guess formulae with cyclic dependencies only become a problem when they aren't bottled.

Reply:
Yeh, would be good to error out in the same way as `FormulaInstaller` does it here 👍🏻